sap_hana/datadog_checks/sap_hana/sap_hana.py

```diff
@@ -75,6 +76,26 @@ def __init__(self, name, init_config, instances):
         self.check_initializations.append(self.parse_config)
 
     def check(self, instance):
+        default_queries = [
+            self.query_master_database,
+            self.query_database_status,
+            self.query_backup_status,
+            self.query_licenses,
+            self.query_connection_overview,
+            self.query_disk_usage,
+            self.query_service_memory,
+            self.query_service_component_memory,
+            self.query_row_store_memory,
+            self.query_service_statistics,
+            self.query_volume_io,
+            self.query_custom,
+        ]
+
+        if self._only_custom_queries:
+            query_methods = [self.query_custom]
+        else:
+            query_methods = default_queries
```

Comment:
can this be done in the __init__ since it looks like it doesn't have to be configured at every check run. Perhaps `self.query_methods`?

Reply:
Is it conventional to define the entire list of default methods in the `init`? This works, but it doesn't seem like the right place to define the list since it's not configuration.

``` 
def __init__(...):
  self._default_methods = [
    self.query_master_database,
    ...
  ]
```

I also tried this, but the check wasn't able to make each of the method calls. I'm trying to debug why this isn't working.

```
def __init__(...):
  self._default_methods = []
  ...
  self.check_initializations.append(self.set_default_methods)
...

def set_default_methods(self):
  self._default_methods.extend([<list of methods>])
```

Reply:
I think I found the issue. The unit tests asserting for errors weren't getting the errors propagated back so they were failing. I updated them to use `dd_run_check`.